BibBuddy/app/src/main/java/de/bibbuddy/HomeFragment.java

```diff
@@ -1,38 +1,151 @@
+    context = view.getContext();
 
     MainActivity mainActivity = (MainActivity) requireActivity();
     mainActivity.setVisibilityImportShareButton(View.GONE, View.GONE);
     mainActivity.setVisibilitySortButton(false);
 
     mainActivity.updateHeaderFragment(getString(R.string.navigation_home));
     mainActivity.updateNavigationFragment(R.id.navigation_home);
+    updateWelcomeMessage();
+
+    bookModel = new BookModel(context, 1L);
+    dbHelper = new DatabaseHelper(context);
+    bookDao = new BookDao(dbHelper);
+
+    bookAmount = 3;
+    List<Book> bookList = bookDao.findModifiedBooks(bookAmount);
+    bookItemList = getBookItems(bookList);
+
+    adapter = new BookRecyclerViewAdapter(bookItemList, this, getContext());
+
+    setupBooksRecyclerView(bookItemList);
 
     return view;
   }
 
+  private void setupBooksRecyclerView(List<BookItem> bookItemList) {
+    RecyclerView recyclerView = view.findViewById(R.id.home_books_list);
+    adapter = new BookRecyclerViewAdapter(bookItemList, this, getContext());
+    recyclerView.setAdapter(adapter);
+
+    updateEmptyView(bookItemList);
+  }
+
+  private void updateEmptyView(List<BookItem> bookList) {
+    TextView emptyView = view.findViewById(R.id.home_books_list_empty);
+    if (bookList.isEmpty()) {
+      emptyView.setVisibility(View.VISIBLE);
+    } else {
+      emptyView.setVisibility(View.GONE);
+    }
+  }
+
+  @Override
+  public void onBookClicked(int position) {
+    BookItem bookItem = bookItemList.get(position);
+
+    BookNotesView fragment = new BookNotesView();
+    fragment.setArguments(createBookBundle(bookItem));
+
+    showFragment(fragment);
+  }
+
+  @Override
+  public void onBookLongClicked(int position, BookItem bookItem, View v) {
+  }
+
+  private Bundle createBookBundle(LibraryItem item) {
+    Bundle bundle = new Bundle();
+
+    Long currentBookId = item.getId();
+    String currentBookTitle = item.getName();
+
+    bundle.putLong(LibraryKeys.SHELF_ID, bookDao.findShelfIdByBook(item.getId()));
+    bundle.putString(LibraryKeys.SHELF_NAME, bookDao.findShelfNameByBook(item.getId()));
+
+    bundle.putLong(LibraryKeys.BOOK_ID, currentBookId);
+    bundle.putString(LibraryKeys.BOOK_TITLE, currentBookTitle);
+
+    return bundle;
+  }
+
+  private List<BookItem> getBookItems(List<Book> bookList) {
+    List<BookItem> bookItemList = new ArrayList<BookItem>();
+
+    for (Book book : bookList) {
+      bookModel.setShelfId(bookDao.findShelfIdByBook(book.getId()));
+
+      bookItemList.add(
+          new BookItem(
+              book,
+              bookDao.findShelfIdByBook(book.getId()),
+              bookModel.getAuthorString(book.getId()),
+              bookDao.countAllNotesForBook(book.getId())
+          )
+      );
+    }
+
+    return bookItemList;
+  }
+
+  private void updateWelcomeMessage() {
+    TextView welcomeMessage = view.findViewById(R.id.welcome_msg);
+
+    switch (ThreadLocalRandom.current().nextInt(1, 5)) {
```

Comment:
```suggestion
```

Reply:
Replace your switch-case code snippet with an if-clause construction (that is what google style format would suggest here and we did it everywhere else in our code - I know it is a bit tricky right now since the checkstyle-plugin does not work right now.
int i = ThreadLocalRandom.current().nextInt(1, 5);
    if (i == 2) {
      welcomeMessage.setText(R.string.welcome_2);
    } else if (i == 3) {
      welcomeMessage.setText(R.string.welcome_3);
    } else if (i == 4) {
      welcomeMessage.setText(R.string.welcome_4);
    } else {
      welcomeMessage.setText(R.string.welcome_1);
    }

Reply:
Thanks, done!